compiler/src/codegen/transl_anf.re

```diff
@@ -1048,36 +1058,55 @@ let lift_imports = (env, imports) => {
     | GrainValue(mod_, name) =>
       let mimp_mod = Ident.create_persistent(mod_);
       let mimp_name = Ident.create_persistent(name);
-      let alloc =
+      let import_name = Printf.sprintf("gimport_%s_%s", mod_, name);
+      let (alloc, mods) =
         switch (imp_shape) {
-        | GlobalShape(alloc) => alloc
+        | GlobalShape(alloc) => (
+            alloc,
+            [
+              {
+                mimp_mod,
+                mimp_name,
+                mimp_type: process_shape(true, imp_shape),
+                mimp_kind: MImportGrain,
+                mimp_setup: MCallGetter,
+                mimp_used: true,
+              },
+            ],
+          )
         | FunctionShape(_) =>
```

Comment:
Can you explain FunctionShape to me a bit? Since it seems like we're adding support that we didn't have before

Reply:
Of course! Imports take one of two shapes, `GlobalShape` or `FunctionShape`, corresponding to wasm globals or wasm functions. Imports that happen via `foreign wasm` are either `GlobalShape` or `FunctionShape`, and before this PR, Grain imports were only `GlobalShape`, meaning that imported functions were only ever imported as values—the closure for the function. This meant that the only way the function could be called was via a `call_indirect`. 

This change allows us say that we're importing an actual Grain webassembly function (and its closure), so we can call it via a regular `call` instruction rather than _having_ to pull the pointer from the closure.